src/hist/basehist.py

```diff
@@ -320,7 +325,21 @@ def plot1d(
 
         import hist.plot
 
-        return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        if self.ndim == 1:
+            return hist.plot.histplot(self, ax=ax, **_proc_kw_for_lw(kwargs))
+        else:
+            if overlay is not None:
+                (cat_ax,) = [ax for ax in self.axes if ax.name == overlay]
+            else:
+                (cat_ax,) = [ax for ax in self.axes if ax.traits.discrete]
+            if cat_ax.traits.discrete:
+                cats = [cat_ax.value(i) for i in range(len(cat_ax))]
+            else:
+                cats = [bh.loc(cat_ax.value(i)) for i in range(len(cat_ax))]
+            d1hists = [self[{cat_ax.name: cat}] for cat in cats]
+            return hist.plot.histplot(
+                d1hists, ax=ax, label=cats, **_proc_kw_for_lw(kwargs)
+            )
```

Comment:
Sourcery probably wants this to be "guard" style, where you remove the else and unindent the block. I'm mildly in favor, as less indentation is nicer, and I'm used to guard style from Ruby (where it is quite common and tied into the syntax a bit more).